services/slam/builtin/builtin.go

```diff
@@ -370,7 +370,10 @@ func setupGRPCConnection(ctx context.Context, port string, logger golog.Logger)
 	// The 'port' provided in the config is already expected to include "localhost:", if needed, so that it doesn't need to be
 	// added anywhere in the code. This will allow cloud-based SLAM processing to exist in the future.
 	// TODO: add credentials when running SLAM processing in the cloud.
-	connLib, err := grpc.DialContext(ctx, port, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithBlock())
+	maxMsgSize := 32 * 1024 * 1024
+	maxMsgSizeOption := grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMsgSize))
```

Comment:
[opt] I'd put `32 * 1024 * 1024` directly into the function call and remove the `maxMsgSize` variable, but I'd also add a comment saying why we're increasing the message size from 4MB to 32MB (see slam repo for an example).